lesson01.calculator/app.js

```diff
@@ -0,0 +1,387 @@
+    }
+    
+    
+    //calculation the arithmetic expression
+    let result;
+    
+    for (let i = elementsHistory.length - 1; i >= 0 ; i--) {
+        if(elementsHistory[i] === '/' ||  elementsHistory[i] === '*'){
+            if(elementsHistory[i] === "/"){
+                if(elementsHistory[i-1] === 0){
+                    elementsHistory=[];
+                    document.querySelector('.screen').innerHTML = 'ERORR';
+                    screen='';
+                    element='';
+                    return;
+                }
+                else    
+                    result = elementsHistory[i+1] / elementsHistory[i-1];
+            }
+            else if(elementsHistory[i] === '*')
+                result = elementsHistory[i+1] * elementsHistory[i-1];
+            
+            elementsHistory[i+1] = result;
+            elementsHistory.splice(i-1, 2);
+            i = elementsHistory.length -1;
+        }
+    }
+    
+    for (let i = elementsHistory.length - 1; i >= 0 ; i--) {
+        if(elementsHistory[i] === '+' ||  elementsHistory[i] === '-'){
+            if(elementsHistory[i] === '+')
+                result = elementsHistory[i+1] + elementsHistory[i-1];
+            else if(elementsHistory[i] === '-')
+                result = elementsHistory[i+1] - elementsHistory[i-1];
+
+            elementsHistory[i+1] = result;
+            elementsHistory.splice(i-1, 2);
+            i = elementsHistory.length - 1;
+        }
+    }
+    //end of the calculation
+
+
+    elementsHistory[0] = elementsHistory[0].toFixed(10);
+    elementsHistory[0] = parseFloat(elementsHistory[0]);
+    elementsHistory[0] = elementsHistory[0].toString();
+    document.querySelector('.screen').innerHTML = elementsHistory[0];
+    screen = elementsHistory[0];
+    element = '';
+    showingResultFlag = true;                               
+});
+
+
+
+
+//----------------Event Listeners for all the special buttons----------//
+document.querySelector('.point').addEventListener('click', function(){
+    if(screen.length === 0)
+        return;
+    
+    if(screen[screen.length-1] === '.')
+        return;
+    
+    if(showingResultFlag === true)
+        return;
+    
+    //loop that check if the last user action wat a arithmetic operations. if it was a point can't be clicked
+    for(let i = 0; i<=operations.length-1; i++){
+        if(screen[screen.length-1]===operations[i])
+            return;
+    }
+    
+    screen += document.querySelector('.point').innerHTML;
+    element += document.querySelector('.point').innerHTML;
+    elementsHistory[0] = element;
+    document.querySelector('.screen').innerHTML = screen; 
+});
+
+
+document.querySelector('.clear').addEventListener('click', function(){
+    screenCheck();
+    screen = '';
+    document.querySelector('.screen').innerHTML = screen;
+    elementsHistory=[];
+    element = '';
+});
+
+
+document.querySelector('.delete').addEventListener('click', function(){
+    if(screen.length === 0){
```

Comment:
see my last comment - try to make it [simple and short](https://en.wikipedia.org/wiki/KISS_principle)